mlir/lib/Analysis/Presburger/IntegerRelation.cpp

```diff
@@ -1946,9 +1944,8 @@ void IntegerRelation::fourierMotzkinEliminate(unsigned pos, bool darkShadow,
   for (unsigned r = 0, e = getNumEqualities(); r < e; r++) {
     if (atEq(r, pos) != 0) {
       // Use Gaussian elimination here (since we have an equality).
-      LogicalResult ret = gaussianEliminateVar(pos);
-      (void)ret;
-      assert(succeeded(ret) && "Gaussian elimination guaranteed to succeed");
+      bool ret = gaussianEliminateVar(pos);
```

Comment:
You still need the `(void) ret` here, or `[[maybe_unused]]`, otherwise it will warn in release builds.